pkg/validate/pod.go

```diff
@@ -196,3 +313,76 @@ func createPodSandboxWithLogDirectory(c internalapi.RuntimeService) (sandboxID s
 
 	return framework.RunPodSandbox(c, podConfig), podConfig, hostPath
 }
+
+// testMetricDescriptors verifies that all expected metric descriptors are present.
+func testMetricDescriptors(descs []*runtimeapi.MetricDescriptor) {
+	returnedDescriptors := make(map[string]*runtimeapi.MetricDescriptor)
+	for _, desc := range descs {
+		returnedDescriptors[desc.GetName()] = desc
+		Expect(desc.GetHelp()).NotTo(BeEmpty(), "Metric descriptor %q should have help text", desc.GetName())
+		Expect(desc.GetLabelKeys()).NotTo(BeEmpty(), "Metric descriptor %q should have label keys", desc.GetName())
+	}
+
+	missingMetrics := []string{}
+
+	for _, expectedName := range expectedMetricDescriptorNames {
+		_, found := returnedDescriptors[expectedName]
+		if !found {
+			missingMetrics = append(missingMetrics, expectedName)
+		}
+	}
+
+	Expect(missingMetrics).To(BeEmpty(), "Expected %s metrics to be present and they were not", strings.Join(missingMetrics, " "))
+}
+
+// listPodSandboxMetrics lists PodSandboxMetrics.
+func listPodSandboxMetrics(c internalapi.RuntimeService) []*runtimeapi.PodSandboxMetrics {
+	By("List PodSandboxMetrics.")
+
+	metrics, err := c.ListPodSandboxMetrics(context.TODO())
+	framework.ExpectNoError(err, "failed to list PodSandboxMetrics: %v", err)
+	framework.Logf("List PodSandboxMetrics succeed")
+
+	return metrics
+}
+
+// testPodSandboxMetrics verifies that metrics are present for the specified pod.
+func testPodSandboxMetrics(allMetrics []*runtimeapi.PodSandboxMetrics, podID string) {
```

Comment:
Are still taking in this function. Only the metrics descriptors test are needed right?